### PR TITLE
obs-scripting: Remove Python version upper limit on Linux

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Remove Python version upper limit on Linux

### Motivation and Context

We don't rely on the stable ABI on Linux…

### How Has This Been Tested?

Quickly tested the url-text.py script.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by RytoEX — APPROVED** [review]
Looks okay at a glance. @PatTheMav ?

File: deps/obs-scripting/cmake/python.cmake [hunk 1]
```diff
@@ -7,6 +7,8 @@ if(ENABLE_SCRIPTING_PYTHON)
 
   if(OS_WINDOWS)
     find_package(Python 3.8...<3.11 REQUIRED Interpreter Development)
+  elseif(OS_LINUX)
```
Comment (PatTheMav) [inline deps/obs-scripting/cmake/python.cmake:10]: Probably should include FreeBSD and OpenBSD here as well as neither uses runtime linking.

File: deps/obs-scripting/obspython/CMakeLists.txt [hunk 2]
```diff
@@ -14,7 +14,11 @@ if(POLICY CMP0094)
   cmake_policy(SET CMP0094 NEW)
 endif()
 
-find_package(Python 3.8...<3.12 REQUIRED Interpreter Development)
+if(OS_LINUX)
```
Comment (PatTheMav) [inline deps/obs-scripting/obspython/CMakeLists.txt:17]: See above.

File: deps/obs-scripting/cmake/python.cmake (hunk 1 above)
Comment (tytan652) [reply deps/obs-scripting/cmake/python.cmake]: Done

**Review by PatTheMav — APPROVED** [review]

**Pull request merged**